server.js

```diff
@@ -20,4 +20,57 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
+app.get('/messages', function (request, response) {
+  response.json(messages)
+})
+
+app.post('/messages', function (request, response) {
+  let newMessage = {};
+
+  if (request.body.from.trim() && request.body.text.trim()) {
+    newMessage.id = messages.length
```

Comment:
There is a problem with using the length of the message array to assign an ID. For example, if you has a message array of 3 messages, with IDs of 0,1,2, and you deleted message 1, then created a new message it would have an ID of 2. This would mean that there would be 2 messages with ID 2, and IDs have to be unique.

Reply:
```suggestion
    newMessage.id = messages[messages.length - 1].id + 1
```
I think this should solve the problem, don't you think?